corehq/apps/receiverwrapper/views.py

```diff
@@ -73,43 +73,89 @@ def _process_form(request, domain, app_id, user_id, authenticated,
         'domain': domain
     }
 
+    instance, attachments, response = _get_instance_and_attachments(
+        request, domain, authenticated, metric_tags, app_id, user_id)
+
+    if response:
+        return response
+
+    with TimingContext() as timer:
+        app_id, build_id = get_app_and_build_ids(domain, app_id)
+        submission_post = _submission_post_object(
+            instance=instance, attachments=attachments, domain=domain, app_id=app_id, build_id=build_id,
+            auth_cls=auth_cls, user_id=user_id, authenticated=authenticated, request=request)
+
+        try:
+            result = submission_post.run()
+        except XFormLockError as err:
+            return _xform_lock_error_response(
+                domain, authenticated, request, err, metric_tags, app_id,
+                user_id)
+
+    response = result.response
+    _record_metrics(metric_tags, result.submission_type, result.response, timer, result.xform)
+
+    return response
+
+
+def _get_instance_and_attachments(request, domain, authenticated,
+                                  metric_tags, app_id, user_id):
+    response, instance, attachments = None, None, None
     try:
         instance, attachments = couchforms.get_instance_and_attachment(request)
     except MultimediaBug:
-        try:
-            instance = request.FILES[MAGIC_PROPERTY].read()
-            xform = convert_xform_to_json(instance)
-            meta = xform.get("meta", {})
-        except:
-            meta = {}
-
-        metrics_counter('commcare.corrupt_multimedia_submissions', tags={
-            'domain': domain, 'authenticated': authenticated
-        })
-        return _submission_error(
-            request, "Received a submission with POST.keys()", metric_tags,
-            domain, app_id, user_id, authenticated, meta,
-        )
+        response = _multimedia_bug_response(request, domain, authenticated,
+                                            metric_tags, app_id, user_id)
+    else:
+        if isinstance(instance, BadRequest):
+            response = _bad_request_response(instance.message, metric_tags)
```

Comment:
similar comment to https://github.com/dimagi/commcare-hq/pull/28304/files#r477092516 I think it makes more sense to raise exceptions for these errors

Reply:
Dropped a note offline, I'd like to avoid making changes to that since `get_instance_and_attachment` is used at multiple places. All of the changes in this PR are just for better readability without really changing anything.
I am happy to take that as a followup later when I have some more bandwidth.

Reply:
refactor to change to raising exceptions here: https://github.com/dimagi/commcare-hq/pull/28429 so this will need to be updated